constraints/src/main/java/gov/nasa/jpl/aerie/constraints/tree/RollingThreshold.java

```diff
@@ -0,0 +1,120 @@
+package gov.nasa.jpl.aerie.constraints.tree;
+
+import gov.nasa.jpl.aerie.constraints.model.ConstraintResult;
+import gov.nasa.jpl.aerie.constraints.model.EvaluationEnvironment;
+import gov.nasa.jpl.aerie.constraints.model.LinearEquation;
+import gov.nasa.jpl.aerie.constraints.model.LinearProfile;
+import gov.nasa.jpl.aerie.constraints.model.SimulationResults;
+import gov.nasa.jpl.aerie.constraints.model.Violation;
+import gov.nasa.jpl.aerie.constraints.time.Interval;
+import gov.nasa.jpl.aerie.constraints.time.Segment;
+import gov.nasa.jpl.aerie.constraints.time.Spans;
+import gov.nasa.jpl.aerie.constraints.time.Windows;
+import gov.nasa.jpl.aerie.merlin.protocol.types.Duration;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+public record RollingThreshold(Expression<Spans> spans, Expression<Duration> width, Expression<Duration> threshold, RollingThresholdAlgorithm algorithm) implements Expression<ConstraintResult> {
+
+  public enum RollingThresholdAlgorithm {
+    ExcessSpans,
+    ExcessHull,
+    DeficitSpans,
+    DeficitHull
+  }
+
+  @Override
+  public ConstraintResult evaluate(SimulationResults results, final Interval bounds, EvaluationEnvironment environment) {
+    final var width = this.width.evaluate(results, bounds, environment);
+    var spans = this.spans.evaluate(results, bounds, environment);
+
+    final Spans reportedSpans;
+    if (algorithm == RollingThresholdAlgorithm.ExcessHull || algorithm == RollingThresholdAlgorithm.ExcessSpans) {
+      reportedSpans = spans;
+    } else {
+      reportedSpans = spans.intoWindows().not().intoSpans(bounds);
+    }
```

Comment:
Are there more possible RollingWindows Algorithms? If so, I think a switch where the `default` case throws like what is done elsewhere in the codebase would make sense here and on L59

Reply:
Yes, there probably are. I've added an else block that catches this.